frontend/src/components/Map.tsx

```diff
@@ -1,82 +1,101 @@
-
-const GOLDEN: Region = {
-  latitude: 39.749675,
-  longitude: -105.222606,
-  latitudeDelta: 0.005,
-  longitudeDelta: 0.005
-}
+import { LocationButton } from './LocationButton'
+import React, { useRef, useMemo, useState } from 'react'
 
 /**
- * Wraps the expo {@interface MapView} with additional functionality.
+ * A wrapper around react native {@class MapView} that provides a simplified interface for the purposes of this app.
  */
-export function Map(props: ViewProps & MapProps): React.ReactElement<ViewProps> {
-  const [userRegionChanged, setUserRegionChanged] = useState(false)
+export function Map(props: MapProps & ViewProps): React.ReactElement<MapProps & ViewProps> {
   const mapRef = useRef<MapView>(null)
+  const [followingLocation, setFollowingLocation] = useState<boolean>(true)
+  const [lastLocation, setLastLocation] = React.useState<Coordinate | undefined>(undefined)
 
   function panToLocation(location: Coordinate | undefined): void {
-    // We want to make sure we won't snap back to the user location if they decide to pan around,
-    // so check if that's the case before panning.
-    if (location !== undefined && mapRef.current != null && !userRegionChanged) {
+    if (location !== undefined && mapRef.current != null) {
       mapRef.current.animateCamera({
         center: location,
         zoom: 17
       })
     }
   }
 
-  function handleRegionChange(details: Details): void {
-    // If the user is panning around, we don't want to snap back to their location. Note that we
-    // make sure to exclude camera pans from this to avoid disabling location following at soon
-    // as it changes.
-    // details.isGesture is only unavailable on apple maps (which we aren't using)
-    // We default to true because we don't want to lock the map if something is broken
-    if (details.isGesture ?? true) {
-      setUserRegionChanged(true)
+  function updateLocation(location: Coordinate | undefined): void {
+    // Required if we need to re-toggle location following later, which requires
+    // us to pan the camera to whatever location we were last given. Always track
+    // this regardless of if following is currently on.
+    setLastLocation(location)
+    if (followingLocation) {
+      panToLocation(location)
     }
   }
 
+  function flipFollowingLocation(): void {
+    const newState = !followingLocation
+    setFollowingLocation(newState)
+    // There will be some delay between the next location update and when the
+    // location button was toggled, so we need to immediately pan now with the
+    // last location we got.
+    if (newState) {
+      panToLocation(lastLocation)
+    }
+  }
+
+  // Combine given insets with the status bar height to ensure that the map
+  // is fully in-bounds.
   const statusBarInset = useMemo(() => StatusBar.currentHeight ?? 0, [])
   const padding = {
-    // We always inset by the status bar regardless of the user configuration.
     top: props.insets?.left ?? 0 + statusBarInset,
```

Comment:
I think you might want parens around  (props.insets?.left ?? 0) to ensure the math is correct... also the prop left inset is here in the top prop... is that right?